### PR TITLE
Move options from RegexComponent to Regex

### PR DESCRIPTION
**Pull request by Azoy**

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please test

File: Tests/RegexBuilderTests/RegexDSLTests.swift [hunk 1]
```diff
@@ -307,11 +315,13 @@ class RegexDSLTests: XCTestCase {
...
-          OneOrMore(.word)
-          Anchor.wordBoundary
-        }
-        .wordBoundaryKind(.unicodeLevel1)
+        Regex {
+          Capture {
+            OneOrMore(.word)
+            Anchor.wordBoundary
+          }
+        }.wordBoundaryKind(.unicodeLevel1)
```
Comment (milseman) [inline Tests/RegexBuilderTests/RegexDSLTests.swift:323]: This will need to be updated, depending on merge order of PRs

**Review by milseman — APPROVED** [review]
LGTM

**Comment by Azoy** [comment]
@swift-ci please test

**Pull request merged**